django/db/migrations/autodetector.py

```diff
@@ -905,6 +918,36 @@ def generate_altered_fields(self):
                     self._generate_removed_field(app_label, model_name, field_name)
                     self._generate_added_field(app_label, model_name, field_name)
 
+    def generate_altered_indexes(self):
+        option_name = "indexes"
+        for app_label, model_name in sorted(self.kept_model_keys):
+            old_model_name = self.renamed_models.get((app_label, model_name), model_name)
+            old_model_state = self.from_state.models[app_label, old_model_name]
+            new_model_state = self.to_state.models[app_label, model_name]
+            new_model = self.to_state.apps.get_model(app_label, model_name)
+
+            old_indexes = old_model_state.options.get(option_name)
+            new_indexes = new_model_state.options.get(option_name)
+            add_diff = [idx for idx in new_indexes if idx not in old_indexes]
```

Comment:
I had tried it with `set` arithmetics before implementing this but it didn't seem to work.
Although `old_indexes == new_indexes` evaluated to `True`,
`set(old_indexes) == set(new_indexes)` evaluated to `False`. Do I need to implement `__hash__` for the `Index` class for it to work.
I didn't put more efforts on this because not using sets worked for me, but let me know if I should look deeper into this matter.